src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -229,7 +229,8 @@ public CheckResult Check(ParseResult parse, RecordType parameterType, ParserOpti
                 CultureInfo = Config.CultureInfo
             };
 
-            if (result.IsSuccess)
+            // If the result is not sucess or has warning related to DeferredArgs we don't want to calculate return type or generate the IR Tree.
+            if (result.IsSuccess && !result.HasDeferredArgsWarning)
```

Comment:
>&& !result.HasDeferredArgsWarning [](http://example.com/codeflow?start=33&length=33)

Note that PowerApps will call the binder directly and not go through this method. 
And we want to ensure that we're not changing error cases into passing cases. 

So we need to be sure that either:
1. Any check that we "loosened" to no longer create an error ("X.field") was a case that didn't happen in Power Apps. 
2. Have a binder flag so that PowerApps issues error. 
3. Possibly post-process - always add the error (so PowerApps gets it), but then remove error in this path.

Reply:
I have now added the flag to mitigate the possible risk to PowerApps.